sql/generate.sql

```diff
@@ -0,0 +1,308 @@
+    WITH threads AS MATERIALIZED (
+        SELECT array_agg(threads.id) AS ids
+        FROM cards AS threads
+        WHERE threads.type = 'thread@1.0.0'
+        ORDER BY random()
+    )
+    INSERT INTO links2 (id, forward, fromid, name, toid)
+    SELECT
+        uuid_in(md5(random()::text || clock_timestamp()::text)::cstring),
+        true,
+        threads.ids[(random() * (array_length(threads.ids, 1) - 1) + 1)::int],
+        (SELECT id FROM strings WHERE string='has attached element'),
+        cards.id
+    FROM cards, threads
+    WHERE cards.type = 'message@1.0.0';
+
+    WITH threads AS MATERIALIZED (
+        SELECT array_agg(threads.id) AS ids
+        FROM cards AS threads
+        WHERE threads.type = 'thread@1.0.0'
+        ORDER BY random()
+    )
+    INSERT INTO links2 (id, forward, fromid, name, toid)
+    SELECT
+        uuid_in(md5(random()::text || clock_timestamp()::text)::cstring),
+        false,
+        cards.id,
+        (SELECT id FROM strings WHERE string='is attached to'),
+        threads.ids[(random() * (array_length(threads.ids, 1) - 1) + 1)::int]
+    FROM cards, threads
+    WHERE cards.type = 'message@1.0.0';
+
+    -- Seed 1m action-request cards
+    RAISE INFO 'Creating action-request contracts...';
+    INSERT INTO cards (id, slug, type, active, version_major, version_minor, version_patch, tags, markers, links, requires, capabilities, data, linked_at, created_at)
+    SELECT
+        uuid_in(md5(random()::text || clock_timestamp()::text)::cstring),
+        'action-request-' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring),
+        'action-request@1.0.0',
+        random() > .01,
+        1,
+        0,
+        0,
+        '{}',
+        '{}',
+        '{}',
+        '{}',
+        '{}',
+        ('{"actor": "' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '", "epoch": ' || extract(EPOCH FROM (NOW() + (random() * (interval '90 days')) + '30 days'))::integer || ', "input": {"id": "' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '"}, "action": "action-create-card@1.0.0", "context": {"id": "REQUEST-28.0.3-' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '", "api": "SERVER-28.0.3-localhost-' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '"}, "arguments": {"reason": null, "properties": {"data": {"actor": "' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '", "expiration": "' || (NOW() + (random() * (interval '90 days')) + '30 days')::text || '"}, "slug": "session-ui-user-johndoe-' || uuid_in(md5(random()::text || clock_timestamp()::text)::cstring) || '", "linked_at": {}}}, "timestamp": "' || (NOW() + (random() * (interval '90 days')) + '30 days')::text || '"}')::json,
+        '{}',
+        (NOW() + (random() * (interval '90 days')) + '30 days')
+    FROM generate_series(1, 1000000);
+
+    -- Create exactly one execute card for each action-request
```

Comment:
Obviously shouldn't merge this as-is since `execute` contracts no longer exist on clean systems. Keeping it here for now to generate data necessary for testing the archive SQL script.